servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/GracefulConnectionClosureHandlingTest.java

```diff
@@ -241,26 +268,57 @@ public Completable accept(final ConnectionContext context) {
         toClose = initiateClosureFromClient ? connection : serverContext;
     }
 
-    @SuppressWarnings("unused")
+    private static HttpProtocolConfig protocolConfig(HttpProtocol protocol, boolean withKeepAlive) {
+        switch (protocol) {
+            case HTTP_1:
+                return protocol.config;
+            case HTTP_2:
+                return applyFrameLogger(h2())
+                        .keepAlivePolicy(withKeepAlive ?
+                                whenIdleFor(ofMillis(TIMEOUT_MILLIS + 200), ofMillis(TIMEOUT_MILLIS)) : disabled())
+                        .build();
+            default:
+                throw new IllegalArgumentException("Unknown protocol: " + protocol);
+        }
+    }
+
     private static Collection<Arguments> data() {
         Collection<Arguments> data = new ArrayList<>();
         for (HttpProtocol protocol : values()) {
-            for (boolean initiateClosureFromClient : TRUE_FALSE) {
-                for (boolean useUds : TRUE_FALSE) {
-                    for (boolean viaProxy : TRUE_FALSE) {
-                        if (viaProxy && (useUds || protocol == HTTP_2)) {
-                            // UDS cannot be used via proxy
-                            // Proxy is not supported with HTTP/2
-                            continue;
+            for (boolean secure : TRUE_FALSE) {
+                for (boolean initiateClosureFromClient : TRUE_FALSE) {
+                    for (boolean useUds : TRUE_FALSE) {
+                        for (boolean viaProxy : TRUE_FALSE) {
+                            if (viaProxy && (useUds || protocol == HTTP_2 || !secure)) {
+                                // UDS cannot be used via proxy
+                                // Proxy is not supported with HTTP/2
+                                // Proxy tunnel works only with secure connections
+                                continue;
+                            }
+                            data.add(Arguments.of(protocol, secure, initiateClosureFromClient, useUds, viaProxy));
                         }
-                        data.add(Arguments.of(protocol, initiateClosureFromClient, useUds, viaProxy));
                     }
                 }
             }
         }
         return data;
     }
 
+    private static Collection<Arguments> dataWithKeepAlive() {
```

Comment:
consider adding a docblock here, the method is quite dense with array manipulation - just briefly explaining what it does to the arguments?